boards: npcx: add support for npck3m8k_evb

alvsun opened the pull request:
Add support for npck3m8k board that is a development platform to evaluate the Nuvoton NPCK3 embedded controller.

nordicjm commented on boards/nuvoton/npck3m8k_evb/Kconfig.defconfig:
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2025 Nuvoton Technology Corporation.
```
remove "arm:" from git, that is hwmv1 format

nordicjm commented on boards/nuvoton/npck3m8k_evb/doc/index.rst:
```diff
@@ -0,0 +1,96 @@
+.. _npck3m8k_evb:
+
+NPCK3M8K_EVB
+###################
```
mismatched length

nordicjm commented on boards/nuvoton/npck3m8k_evb/doc/index.rst:
```diff
@@ -0,0 +1,96 @@
+Supported Features
+==================
+
+The following features are supported:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| CLOCK     | on-chip    | reset and clock control             |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not currently supported by Zephyr (at the moment)
+
+The default configuration can be found in the defconfig file:
+``boards/arm/npck3m8k_evb/npck3m8k_evb_defconfig``
```
use hw directive

kartben replied:
Yes. Please :) in fact, please check the board template and make sure you're following it.
https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/templates/board.tmpl

alvsun replied:
Done, thanks

nordicjm commented on boards/nuvoton/npck3m8k_evb/doc/npck3m8k_evb.png:
file size way too large, convert to webp, then put through https://tinypng.com/

alvsun replied:
Done

nordicjm commented on boards/nuvoton/npck3m8k_evb/npck3m8k_evb.dts:
```diff
@@ -0,0 +1,140 @@
+
+	leds-pwm {
+		compatible = "pwm-leds";
+		pwm_led0_green: pwm_led_0 {
```
newline on line 42 and separating all child nodes, fix in whole PR

nordicjm commented on boards/nuvoton/npck3m8k_evb/npck3m8k_evb_defconfig:
```diff
@@ -0,0 +1,34 @@
+# Clock configuration
+CONFIG_CLOCK_CONTROL=y
+
+# General Kernel Options
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=15000000
+
```
must be set in Kconfig.defconfig file, ideally in soc one, and must come from dts value

alvsun replied:
Done

nordicjm commented on boards/nuvoton/npck3m8k_evb/npck3m8k_evb_defconfig:
```diff
@@ -0,0 +1,34 @@
+# GPIO Driver
+CONFIG_GPIO=y
+
+# Pin Controller Driver
+CONFIG_PINCTRL=y
```
not allowed

alvsun replied:
Done

kartben commented on boards/nuvoton/npck3m8k_evb/doc/index.rst:
```diff
@@ -0,0 +1,96 @@
+.. _npck3m8k_evb:
+
+NPCK3M8K_EVB
+###################
```
This should actually be as follows anyway:


```suggestion
.. zephyr:board:: npck3m8k_evb
```

alvsun replied:
Done

kartben commented on boards/nuvoton/npck3m8k_evb/doc/index.rst:
```diff
@@ -0,0 +1,96 @@
+Nuvoton NPCK3 series microcontrollers. This board needs to be mated with
+part number NPCK3M8K.
+
+.. image:: ./npck3m8k_evb.png
+     :width: 800px
+     :align: center
+     :alt: NPCK3M8K Digital Functionality Tester Board
+
```
zephyr:board:: directive takes care of this 

```suggestion
```

kartben commented on boards/nuvoton/npck3m8k_evb/board.yml:
```diff
@@ -0,0 +1,5 @@
+board:
+  name: npck3m8k_evb
```
Add `full_name` property with commercial name of the board

alvsun replied:
Done

alvsun commented:
The UART error is fixed in [PR #90982](https://github.com/zephyrproject-rtos/zephyr/pull/90982). 
Will rebase after it is merged.

alvsun committed:
boards: npcx: add support for npck3m8k_evb

Add support for npck3m8k board that is a development platform to
evaluate the Nuvoton NPCK3 embedded controller.

Signed-off-by: Alvis Sun <yfsun@nuvoton.com>
Signed-off-by: Mulin Chao <mlchao@nuvoton.com>

nordicjm commented on boards/nuvoton/npck3m8k_evb/Kconfig.defconfig:
```diff
@@ -0,0 +1,8 @@
+config SYS_CLOCK_TICKS_PER_SEC
+	default 1000
+
+config SYS_CLOCK_HW_CYCLES_PER_SEC
+	default 15000000
```
this needs to come from a dts property and using the dts functions, and should be set at SoC level in the Soc Kconfig.defconfig file

nordicjm commented on soc/nuvoton/npcx/npck3/Kconfig.defconfig:
```diff
@@ -15,4 +15,7 @@ config ESPI_TAF_NPCX
 	default y
 	depends on ESPI_TAF
 
+config SYS_CLOCK_HW_CYCLES_PER_SEC
+	default 15000000
```
get from dts property using dts functions

alvsun replied:
Done

alvsun commented:
Does anyone have any additional comments on this PR?

fabiobaltieri commented:
this needs approval from either @MulinChao or @ChiHuaL

kartben commented on boards/nuvoton/npck3m8k_evb/doc/index.rst:
```diff
@@ -0,0 +1,66 @@
+Connections and IOs
+===================
+
+Nuvoton to provide the schematic for this board.
```
not really a fan of this wording - maybe just drop the section for now?

alvsun replied:
Done

kartben reviewed (dismissed):
+1 for docs (comment is non-blocking)

alvsun commented:
> @alvsun @ChiHuaL can you folks please make sure this board is covered in MAINTAINERS file? and maybe have a dedicated area for NPCK too? I don't know enough about it to judge if it deserves its own area

Done (https://github.com/zephyrproject-rtos/zephyr/pull/96191)
Thanks!

kartben commented:
> 
> 
> > @alvsun @ChiHuaL can you folks please make sure this board is covered in MAINTAINERS file? and maybe have a dedicated area for NPCK too? I don't know enough about it to judge if it deserves its own area
> 
> @kartben  Done (https://github.com/zephyrproject-rtos/zephyr/pull/96191)
> Thanks!

Thank you!
